kroxylicious-additional-filters/kroxylicious-schema-validation/src/main/resources/META-INF/services/io.kroxylicious.proxy.filter.FilterFactory

```diff
@@ -0,0 +1 @@
+io.kroxylicious.proxy.filter.schema.ProduceValidationFilter$Factory
```

Comment:
The javadoc for service loaders actually disallows inner classes as service implementations.

https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/ServiceLoader.html#:~:text=discussed%20later.%20The-,type,-must%20be%20public

This restriction seems to have appear between Java 8 and Java 11.  My money is that it relates to modules support.